File: BACKEND/LoginService/src/main/java/com/rentit/signin/entities/City.java
package com.rentit.signin.entities;

import jakarta.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "city")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class City {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "city_id")
    private Integer cityId;

    @Column(name = "city_name", nullable = false)
    private String cityName;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "state_id", nullable = false)
    private State state;

    public City() {}

    public Integer getCityId() {
        return cityId;
    }

    public void setCityId(Integer cityId) {
        this.cityId = cityId;
    }

    public String getCityName() {
        return cityName;
    }

    public void setCityName(String cityName) {
        this.cityName = cityName;
    }

    public State getState() {
        return state;
    }

    public void setState(State state) {
        this.state = state;
    }
}

File: BACKEND/BillingService/BillingService/Models/City.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BillingService.Models
{
    [Table("city")]
    public class City
    {
        [Key]
        [Column("city_id")]
        public int CityId { get; set; }

        [Column("city_name")]
        public string? CityName { get; set; }

        [Column("state_id")]
        public int StateId { get; set; }

        // Navigation property
        [ForeignKey("StateId")]
        public virtual State? State { get; set; }
    }
}
